p2p/src/sync/peer.rs

```diff
@@ -0,0 +1,405 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): A. Altonen
+#![cfg(not(loom))]
+#![allow(unused)]
+use crate::{
+    error::{self, P2pError},
+    event,
+    net::{self, FloodsubService, NetworkService},
+    sync::{self, index, mock_consensus},
+};
+use logging::log;
+use std::collections::HashSet;
+use tokio::sync::mpsc;
+
+/// State of the peer
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum PeerSyncState {
+    /// Peer state is unknown
+    Unknown,
+
+    /// Peer is uploading blocks to local node
+    UploadingBlocks(mock_consensus::BlockHeader),
```

Comment:
Maybe I'm confused and don't understand how this works, but I think this is in the wrong place. The connection state should be irrelevant in the peer state. The problem there is that we're mixing connection state with blockchain state. Now I understand that for an async system this is difficult to manage because we have to know both so that not multiple sync processes execute at the same time, but also these are two different things and probably shouldn't be at the same level.

In other words: Whether "I'm currently doing something for this peer" shouldn't be at the same level of "this peer's tip is this/that block".

Again, maybe there's something I still don't understand.